sql/src/main/java/io/deephaven/sql/LogicalSortAdapter.java

```diff
@@ -0,0 +1,124 @@
+import io.deephaven.qst.table.SortTable;
+import io.deephaven.qst.table.TableSpec;
+import org.apache.calcite.rel.RelFieldCollation;
+import org.apache.calcite.rel.RelFieldCollation.NullDirection;
+import org.apache.calcite.rel.logical.LogicalSort;
+import org.apache.calcite.rel.type.RelDataTypeField;
+import org.apache.calcite.rex.RexInputRef;
+
+import java.util.Objects;
+
+final class LogicalSortAdapter {
+
+    public static TableSpec namedTable(LogicalSort sort, NamedAdapter namedAdapter) {
+        return of(sort, namedAdapter, namedAdapter);
+    }
+
+    public static TableSpec indexTable(LogicalSort sort, IndexRef indexRef) {
+        return of(sort, indexRef, indexRef);
+    }
+
+    private static TableSpec of(LogicalSort sort, RelNodeAdapter relNodeAdapter, FieldAdapter fieldAdapter) {
+        // SQLTODO(sort-offset)
+        if (sort.offset != null) {
+            throw new UnsupportedOperationException("SQLTODO(sort-offset): QST does not support slice...");
+        }
+        final SortTable.Builder builder = SortTable.builder()
+                .parent(relNodeAdapter.table(sort.getInput()));
+        for (RelFieldCollation fieldCollation : sort.collation.getFieldCollations()) {
+            final int inputFieldIndex = fieldCollation.getFieldIndex();
+            final RelDataTypeField inputField = sort.getInput().getRowType().getFieldList().get(inputFieldIndex);
+            final RexInputRef inputRef = new RexInputRef(inputFieldIndex, inputField.getType());
+            final ColumnName inputColumnName = fieldAdapter.input(inputRef, inputField);
+            switch (fieldCollation.direction) {
+                case ASCENDING:
+                    if (fieldCollation.nullDirection == NullDirection.LAST) {
+                        throw new UnsupportedOperationException(
+                                "Deephaven does not currently support ascending sort with nulls last");
+                    }
+                    builder.addColumns(SortColumn.asc(inputColumnName));
+                    break;
+                case DESCENDING:
+                    if (fieldCollation.nullDirection == NullDirection.FIRST) {
+                        throw new UnsupportedOperationException(
+                                "Deephaven does not currently support descending sort with nulls first");
+                    }
+                    builder.addColumns(SortColumn.desc(inputColumnName));
+                    break;
+                default:
+                    throw new UnsupportedOperationException("Unsupported sort direction " + fieldCollation.direction);
+            }
+        }
+        final SortTable sortTable = builder.build();
+        if (sort.fetch != null) {
+            final Expression expression = fieldAdapter.expression(sort, sort.fetch);
+            if (!(expression instanceof Literal)) {
+                throw new UnsupportedOperationException("Deephaven only supports literals for FETCH");
```

Comment:
Are there named params? We should support that, too.

Reply:
Added new feature `SQLTODO(custom-expression)`